### PR TITLE
Update CalculateNetworkFee

### PR DESCRIPTION
**Pull request by belane**
Adjust the calculation of gas used during the verification.

### CODE REVIEW

**Comment by belane** [comment]
@dotnet-policy-service agree company="Belane"

**Review by shargon — DISMISSED** [review]

**Comment by roman-khimov** [comment]
Likely a bit better this way: nspcc-dev/neo-go#3141

File: src/Neo/Wallets/Wallet.cs [hunk 1]
```diff
@@ -596,6 +596,7 @@ public long CalculateNetworkFee(DataCache snapshot, Transaction tx)
             uint exec_fee_factor = NativeContract.Policy.GetExecFeeFactor(snapshot);
             long networkFee = 0;
             int index = -1;
+            long gas_left = ApplicationEngine.TestModeGas;
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply src/Neo/Wallets/Wallet.cs:599]: OK, `settings.MaxGasInvoke` is unreachable from the Wallet class, so resolved.

[the remark this answers is not in the document]
Comment (roman-khimov) [reply src/Neo/Wallets/Wallet.cs:599]: Likely this can be solved with some API extension, a bit more invasive though.

**Review by vncoelho — COMMENTED** [review]
HI @belane, good to see you again.

This is just verification, right?
In this sense, no change in case of resync, yes?

**Comment by belane** [comment]
> HI @belane, good to see you again.
> 
> This is just verification, right? In this sense, no change in case of resync, yes?

Hi @vncoelho , yes it's only for CalculateNetworkFee. It does not affect chain and doesn't need any resync.

**Comment by vncoelho** [comment]
Sometimes need a re-sync is different than being affected by a future re-sync....Just to be sure...aehauhea

**Review by shargon — APPROVED** [review]

**Pull request merged**